From 4c91b27d00664c491e1f7e30d608df6ef74ba3ac Mon Sep 17 00:00:00 2001
From: "Bozek, DominikX" <dominikx.bozek@intel.com>
Date: Tue, 26 Feb 2019 16:10:19 +0100
Subject: [PATCH] fix value of SDxFMT register

The SDxFMT register get malformed, because driver make
a decision on ppcap field in hdac_ext_bus.
Driver maintain ppcap in hdac_bus, not in hdac_ext_bus.

Change-Id: I88d3e2466429419db67ac2b3d6e933c8f7d83fb4
---
 include/sound/hdaudio_ext.h       | 6 ------
 sound/soc/intel/skylake/skl-pcm.c | 3 ++-
 2 files changed, 2 insertions(+), 7 deletions(-)

diff --git a/include/sound/hdaudio_ext.h b/include/sound/hdaudio_ext.h
index ade1b9f..7b1558e 100644
--- a/include/sound/hdaudio_ext.h
+++ b/include/sound/hdaudio_ext.h
@@ -18,12 +18,6 @@ struct hdac_ext_bus {
 	int num_streams;
 	int idx;
 
-	void __iomem *ppcap;
-	void __iomem *spbcap;
-	void __iomem *mlcap;
-	void __iomem *gtscap;
-	void __iomem *drsmcap;
-
 	struct list_head hlink_list;
 
 	struct mutex lock;
diff --git a/sound/soc/intel/skylake/skl-pcm.c b/sound/soc/intel/skylake/skl-pcm.c
index 9018875..fe94353 100644
--- a/sound/soc/intel/skylake/skl-pcm.c
+++ b/sound/soc/intel/skylake/skl-pcm.c
@@ -278,9 +278,10 @@ static int skl_get_format(struct snd_pcm_substream *substream,
 	struct snd_soc_pcm_runtime *rtd = snd_pcm_substream_chip(substream);
 	struct skl_dma_params *dma_params;
 	struct hdac_ext_bus *ebus = dev_get_drvdata(dai->dev);
+	struct hdac_bus *bus = ebus_to_hbus(ebus);
 	int format_val = 0;
 
-	if (ebus->ppcap) {
+	if (bus->ppcap) {
 		struct snd_pcm_runtime *runtime = substream->runtime;
 
 		format_val = snd_hdac_calc_stream_format(runtime->rate,
-- 
1.9.1

